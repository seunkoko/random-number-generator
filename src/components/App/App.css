.App {
  text-align: center;
  word-wrap: break-word;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.home__container {
  margin-top: 4rem;
  height: 6rem;
  color: #222222;
}

.dropdown {
  margin-top: 0.1rem;
  height: 2rem;
}

#numberInput {
  width: 70%;
  font-size: 0.8rem;
  height: 2.5rem;
  padding: 1rem;
}

.btn__dropdown {
  background-color: #F2F2F2;
  width: 40%;
  font-size: 0.8rem;
}

.btn__generate {
  background-color: #222222;
  color: #CCCCCC;
  letter-spacing: 0.2rem;
  font-weight: bold;
  font-size: 0.8rem;
  width: 60%;
  height: 2.5rem;
  word-wrap: break-word !important;
}

.btn__generate:hover {
  background-color: #222222;
  color: #F2F2F2;
}

.btn__generate:focus {
  background-color: #222222;
  color: #F2F2F2;
}

.btn__generate:active {
  background-color: #222222;
  color: #F2F2F2;
}
