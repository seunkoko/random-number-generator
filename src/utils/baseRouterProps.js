// third-party libraries
import PropTypes from 'prop-types';

const baseRouterProps = {
  history: PropTypes.shape({
    action: PropTypes.string,
    block: PropTypes.func,
    createHref: PropTypes.func,
    go: PropTypes.func,
    goBack: PropTypes.func,
    goForward: PropTypes.func,
    length: PropTypes.number,
    listen: PropTypes.func,
    location: PropTypes.shape({
      pathname: PropTypes.string,
      search: PropTypes.string,
    }),
    push: PropTypes.func,
    replace: PropTypes.func,
  }),
  location: PropTypes.shape({
    hash: PropTypes.string,
    pathname: PropTypes.string,
    search: PropTypes.string,
  }),
  match: PropTypes.shape({
    isExact: PropTypes.bool,
    params: PropTypes.object,
    path: PropTypes.string,
    url: PropTypes.string,
  }),
};

export default baseRouterProps;
