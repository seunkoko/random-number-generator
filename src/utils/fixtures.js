export const dummyGeneratedNumbers = [
  { id: 2, value: 2222222222 }, { id: 2, value: 1222222222 }, { id: 1, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 22211122222 }, { id: 2, value: 2222222222 }, { id: 2, value: 22445522222 }, // eslint-disable-line
  { id: 2, value: 2122222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 1, value: 2222222222 }, // eslint-disable-line
  { id: 1, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
  { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, { id: 2, value: 2222222222 }, // eslint-disable-line
];

export const emptyGeneratedNumbers = [];
