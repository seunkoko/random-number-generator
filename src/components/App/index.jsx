import React from 'react';
import logo from '../../assets/images/logo.svg';
import './App.css';

/**
 * App - Landing page
 *
 * @class App
 * @extends {Component}
 */
const App = () => (
  <div className="App">
    <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <h1 className="App-title">Welcome to React</h1>
    </header>
    <p className="App-intro">
      To get started, edit
      <code>
        src/App.js
      </code>
      and save to reload.
    </p>
  </div>
);

export default App;
