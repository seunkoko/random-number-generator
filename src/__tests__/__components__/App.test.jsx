import React from 'react';

// third-party libraries
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { spy } from 'sinon';

// components
import App from '../../components/App';

// utils
import baseRouterProps from '../../utils/baseRouterProps';
import { dummyGeneratedNumbers } from '../../utils/fixtures';

let event = {
  target: {
    name: 'numberToGenerate',
    value: 4000,
    selectedOptions: [
      { innerHTML: 'selected name' },
    ],
  },
  preventDefault: spy(),
};

describe('App', () => {
  const props = {
    ...baseRouterProps,
    homeActive: true,
    operationsActive: true,
    generatedNumbers: dummyGeneratedNumbers,
  };
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App {...props} />);
  });

  it('renders without crashing', () => {
    const rendering = toJson(wrapper);
    expect(rendering).toMatchSnapshot();
  });

  it('should call the onClick method to sort by Highest', () => {
    const action = wrapper.find('a.highestSort').simulate('click');
    expect(action).toBeInstanceOf(Object);
  });

  it('should call the onClick method to sort by Lowest', () => {
    const action = wrapper.find('a.lowestSort').simulate('click');
    expect(action).toBeInstanceOf(Object);
  });

  it('should call the onClick method to sort by IDs', () => {
    const action = wrapper.find('a.uniqueSort').simulate('click');
    expect(action).toBeInstanceOf(Object);
  });

  it('should call the onChange method for input validation', () => {
    const onChangeSpy = spy(wrapper.instance(), 'onChange');
    wrapper.instance().onChange(event);
    wrapper.find('.number').first().simulate('change', event);
    expect(onChangeSpy.called).toEqual(true);
    expect(wrapper.state().numberToGenerate).toEqual(4000);
  });

  it('should call the onChange method', () => {
    event = {
      ...event,
      target: { name: 'dummy' },
    };
    const onChangeSpy = spy(wrapper.instance(), 'onChange');
    wrapper.instance().onChange(event);
    wrapper.find('.number').first().simulate('change', event);
    expect(onChangeSpy.called).toEqual(true);
    expect(wrapper.state().numberToGenerate).toEqual(0);
  });
});
