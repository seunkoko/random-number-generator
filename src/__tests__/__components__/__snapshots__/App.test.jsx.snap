// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App
  generatedNumbers={
    Array [
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 1222222222,
      },
      Object {
        "id": 1,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 22211122222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 22445522222,
      },
      Object {
        "id": 2,
        "value": 2122222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 1,
        "value": 2222222222,
      },
      Object {
        "id": 1,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
      Object {
        "id": 2,
        "value": 2222222222,
      },
    ]
  }
  history={[Function]}
  homeActive={true}
  location={[Function]}
  match={[Function]}
  operationsActive={true}
>
  <div>
    <Header
      homeActive={true}
      operationsActive={false}
    >
      <nav
        className="navbar navbar-inverse"
      >
        <div
          className="container-fluid"
        >
          <div
            className="container header__container"
          >
            <div
              className="navbar-header"
            >
              <img
                alt=""
                className="__logo"
                src="rng-logo-black.png"
              />
              <a
                className="navbar-brand"
                href="/"
              >
                Random Number Generator
              </a>
            </div>
            <ul
              className="nav navbar-nav"
            >
              <li
                className="active"
              >
                <a
                  href="/"
                  id="homeTab"
                >
                  Home
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/operations"
                  id="operationsTab"
                >
                  Operations
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </Header>
    <div
      className="container"
    >
      <div
        className="container home__container"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="col-sm-4 text-left"
            >
              <input
                className="number"
                id="numberInput"
                max="5000"
                min="1"
                name="numberToGenerate"
                onChange={[Function]}
                placeholder="not greater than 5000"
                type="number"
              />
            </div>
            <div
              className="col-sm-4 text-left"
            >
              <div
                className="dropdown"
              >
                <button
                  className="btn btn-default dropdown-toggle btn__dropdown"
                  data-toggle="dropdown"
                  id="menu1"
                  type="button"
                >
                  SORT BY
                  <span
                    className="caret"
                  />
                </button>
                <ul
                  aria-labelledby="menu1"
                  className="dropdown-menu dropdown__container"
                  role="menu"
                >
                  <li
                    role="presentation"
                  >
                    <a
                      className="lowestSort"
                      href="#"
                      onClick={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      Lowest Values
                    </a>
                  </li>
                  <li
                    role="presentation"
                  >
                    <a
                      className="highestSort"
                      href="#"
                      onClick={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      Highest Values
                    </a>
                  </li>
                  <li
                    role="presentation"
                  >
                    <a
                      className="uniqueSort"
                      href="#"
                      onClick={[Function]}
                      role="menuitem"
                      tabIndex="-1"
                    >
                      Unique Identifier
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <div
              className="col-sm-4 text-right"
            >
              <button
                className="btn btn__generate"
                type="submit"
              >
                GENERATE
              </button>
            </div>
          </div>
        </form>
      </div>
      <Table
        generatedNumbers={Array []}
        numberUpdated={false}
      >
        <div>
          <div
            className="table__container container"
          >
            <table
              className="table table-striped no-move"
            >
              <thead
                className="fixed_header"
              >
                <tr>
                  <th>
                    #
                  </th>
                  <th
                    className="text-center"
                  >
                    Unique Identifier
                  </th>
                  <th
                    className="text-center"
                  >
                    Generated Number
                  </th>
                </tr>
              </thead>
              <tbody
                className="scroll_body"
              >
                <tr>
                  <td
                    colSpan="3"
                  >
                    No numbers found
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <Pagination
            handlePageClick={[Function]}
            id="paginateId"
            totalPages={0}
          >
            <nav
              className="container pagination__container"
            />
          </Pagination>
        </div>
      </Table>
    </div>
  </div>
</App>
`;
